server/container_create.go

```diff
@@ -229,13 +229,17 @@ func setupContainerUser(ctx context.Context, specgen *generate.Generator, rootfs
 	}
 
 	genPasswd := true
+	genGroup := true
 	for _, mount := range specgen.Config.Mounts {
 		if mount.Destination == "/etc" ||
 			mount.Destination == "/etc/" ||
```

Comment:
mount.Destination == "/etc" would be a reason to set genGroup to false as well.
I suggest a switch statement:
```
switch mount.Destination {
case "/etc/":
genPasswd, genGroup = false
case "/etc/passwd":
genPasswd = false
case "/etc/group"
genCgroup = false
}
```
you may need a separate function though as break statements won't work in a switch (to break out of the loop). also: don't break out of the loop on only the /etc/passwd case, as a user may have mounted /etc/passwd and /etc/group but this code as-is would only catch the former.

Reply:
Will implement the suggestion

Reply:
For the sake of completeness, the switch statement would be:

```go
	switch mount.Destination {
	case "/etc", "/etc/":
		genPasswd, genGroup = false, false
	case "/etc/passwd":
		genPasswd = false
	case "/etc/group":
		genGroup = false
	}
```